desktop/app/store-helper.js

```diff
@@ -1,10 +1,37 @@
 import {ipcMain} from 'electron'
 
-export default function () {
-  // In case the subscribe store comes before the remote store is ready
+export default function (mainWindow) {
+  const subscribeStoreSubscribers = []
+  let store = {}
+
   ipcMain.on('subscribeStore', event => {
-    ipcMain.on('remoteStoreReady', () => {
-      event.sender.send('resubscribeStore')
+    const sender = event.sender
+    subscribeStoreSubscribers.push(sender)
+
+    try {
+      sender.send('stateChange', store)
+    } catch (_) { }
+  })
+
+  ipcMain.on('stateChange', (event, incomingStore) => {
+    store = incomingStore
+
+    let dead = []
+    subscribeStoreSubscribers.forEach((sub, idx) => {
+      try {
+        sub.send('stateChange', store)
+      } catch (_) {
+        dead.push(idx)
```

Comment:
We keep track of the listeners ourselves so its easier to understand whats going on (vs using the event emitter's internal on/removeListeners mechanism)